src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useRef, useState } from "react";
+
+/**
+ * Custom hook to debounce a value.
+ *
+ * @param {any} value - The value to debounce.
+ * @param {number} delay - The debounce delay in milliseconds.
+ * @returns {any} - The debounced value.
+ */
+const useDebounce = (value: string, delay: number) => {
+  function compareObjects(obj1: string, obj2: string) {
```

Comment:
I observed that two parameters coming in differently but looking the same were being treated as different values. This was causing unnecessary API calls. Upon examining the previous parameter value and the current incoming parameter value, I found they were different. This issue was occurring not only when typing inside the phone number field but also when changing the option, causing unnecessary API calls.

```
{page: 1, limit: 12, name: undefined, patient_no: undefined, is_active: 'True', …}
{page: '1', limit: 12, name: undefined, patient_no: undefined, is_active: 'True', …}
```

If you observe the page key value, the previous one is a number, and the current one is a string. This was the problem. I solved it by comparing two objects while ignoring their value types, which now prevents unnecessary API calls